includes/Checker/Checks/Trademarks_Check.php

```diff
@@ -0,0 +1,543 @@
+				'<code>' . esc_html( $plugin_name ) . '</code>',
+				esc_html( trim( $check, '-' ) )
+			);
+		}
+
+		throw new Exception( $message );
+	}
+
+	/**
+	 * Determines if we find a trademarked term in plugin slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_slug The plugin slug.
+	 *
+	 * @throws Exception Thrown if we found trademarked term in plugin slug.
+	 */
+	private function validate_slug_has_no_trademarks( $plugin_slug ) {
+		if ( empty( $plugin_slug ) ) {
+			return;
+		}
+
+		$check = $this->has_trademarked_slug( $plugin_slug );
+		if ( ! $check ) {
+			return;
+		}
+
+		if (
+			trim( $check, '-' ) === $check
+			&& in_array( $check, self::FOR_USE_EXCEPTIONS, true )
+		) {
+			// Trademarks that do NOT end in "-", but are within the FOR_USE_EXCEPTIONS array can be used, but only if it ends with 'for x'.
+			$message = sprintf(
+				/* translators: 1: plugin slug, 2: found trademarked term */
+				__( 'The plugin slug includes a restricted term. Your plugin slug - %1$s - contains the restricted term "%2$s" which cannot be used within in your plugin slug, unless your plugin slug ends with "for %2$s". The term must still not appear anywhere else in your plugin slug.', 'plugin-check' ),
+				'<code>' . esc_html( $plugin_slug ) . '</code>',
+				esc_html( trim( $check, '-' ) )
+			);
+		} elseif ( trim( $check, '-' ) === $check ) {
+			// Trademarks that do NOT end in "-" indicate slug cannot contain term at all.
+			$message = sprintf(
+				/* translators: 1: plugin slug, 2: found trademarked term */
+				__( 'The plugin slug includes a restricted term. Your plugin slug - %1$s - contains the restricted term "%2$s" which cannot be used at all in your plugin slug.', 'plugin-check' ),
+				'<code>' . esc_html( $plugin_slug ) . '</code>',
+				esc_html( trim( $check, '-' ) )
+			);
+		} else {
+			// Trademarks ending in "-" indicate slug cannot BEGIN with that term.
+			$message = sprintf(
+				/* translators: 1: plugin slug, 2: found trademarked term */
+				__( 'The plugin slug includes a restricted term. Your plugin slug - %1$s - contains the restricted term "%2$s" and cannot be used to begin your plugin slug. We disallow the use of certain terms in ways that are abused, or potentially infringe on and/or are misleading with regards to trademarks. You may use the term "%2$s" elsewhere in your plugin slug, such as "... for %2$s".', 'plugin-check' ),
+				'<code>' . esc_html( $plugin_slug ) . '</code>',
+				esc_html( trim( $check, '-' ) )
+			);
+		}
+
+		throw new Exception( $message );
+	}
+
+	/**
+	 * Whether the plugin uses a trademark in the slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $slug The plugin slug.
+	 * @return string|false The trademark slug if found, false otherwise.
+	 */
+	private function has_trademarked_slug( $slug ) {
+		// We work on slugs for this check.
+		$slug = sanitize_title_with_dashes( $slug );
```

Comment:
I think we should remove this from here, as the method expects a slug anyway. This function should only be called when passing a plugin _name_ to `has_trademarked_slug()`.

Instead, here we could add the check to bail early (with `return false`) if `empty( $slug )`.